invenio_records_resources/services/records/results.py

```diff
@@ -352,13 +353,30 @@ def _find_fields(self, service, value):
 
     def _fetch_referenced(self, grouped_values, identity):
         """Search and fetch referenced recs by ids."""
+
         for service, values in grouped_values.items():
             results = service.read_many(identity, list(values))
+
+            found_values = set()
             for hit in results.hits:
                 value = hit.get("id", None)
+                # keep values visited so we can extract the ones not found due to
+                # possible deletions
+                found_values.add(value)
                 for field in self._find_fields(service, value):
                     field.add_dereferenced_record(service, value, hit)
 
+            not_found_values = values - found_values
+            if not_found_values:
+                for value in not_found_values:
+                    try:
+                        service.read(identity, value)
```

Comment:
maybe we could avoid that by either
1. Assuming that the record is deleted if not found
2. Change resource status to "not_found" rather than "deleted"

Reply:
is worth trying to do the ``read()``...... couldn't this potentially become quite expensive?